service/src/main/java/org/jboss/sbomer/service/feature/sbom/service/RequestEventRepository.java

```diff
@@ -86,6 +87,14 @@ public RequestEventRepository() {
         super(RequestEvent.class);
     }
 
+    @Transactional
+    public RequestEvent updateWithGenericFailure(String requestEventId) {
+        RequestEvent requestEvent = RequestEvent.findById(requestEventId); // NOSONAR
+        requestEvent.setEventStatus(RequestEventStatus.FAILED);
+        requestEvent.setReason(RequestEvent.FAILED_GENERIC_REASON);
```

Comment:
Is there a possibility to add more information here? We catch different exceptions which may provide additional context. I think it would be worth to leverage it.